docs/concepts/immutability.md

```diff
@@ -12,4 +12,81 @@ related:
+         ↓
+      +-----+
+   +--|  C  |-+
+   |  +-----+ |
+   |          |
++-----+    +-----+
+|  A  |    |  B  |
++-----+    +-----+
+"hello"    "world"
+```
+
+If we change the content of `B` to `IPFS!`, all the upstream paths will change as well. In this simple example, the only upstream path is `C`. If we requested content using the pointer, then we would receive different content, but it's important to note that there is no change happening. Node `B` is not being updated. Instead, we are creating a new DAG where the pointer points to CID `E` that points to the concatenation of node `A` and another node `D`.
+
+```text
+   +-----------+
+   |  Pointer  | --------------+
+   +-----------+               |
+                               ↓
+      +-----+               +-----+   
+   +--|  C  |-+         +-- |  E  | --+
+   |  +-----+ |         |   +-----+   |
+   |          |         |             |
++-----+    +-----+     +-----+    +-----+  
+|  A  |    |  B  |     |  A  |    |  D  |
++-----+    +-----+     +-----+    +-----+ 
+"hello"    "world"     "hello"    "IPFS!"  
+```
+
+Again, node `B` does not change. It will always refer to the same content, `world`. Node `A` also appears in the new DAG. This is not because we are _keeping_ it, that would imply the location-addressed paradigm. In the content-addressed system, any time someone writes a block with `"hello"` it will _always_ have CID `A`. This is different to location-addressed systems where we could reuse the original buffer and edit the small substring that represents the difference.
+
+## Website explanation
+
+Here we have a website that displays two headers called `header_1` and `header_2`. The content of the headers is supplied from the variables `string_1` and `string_2`.
+
+```html
+<body>
+    <h1 id="header_1"></h1>
+    <h1 id="header_2"></h1>
+</body>
+<script>
+    let string_1 = "hello";
+    let string_2 = "world";
+    document.getElementById('header_1').textContent=string_1;
+    document.getElementById('header_2').textContent=string_2;
+</script>
+```
+
+The CID of this website is `QmWLdyFMUugMtKZs1xeJCSUKerWd9M627gxjAtp6TLrAgP`. Users can go to [`example.com/QmWLdyFMUugMtKZs1xeJCSUKerWd9M627gxjAtp6TLrAgP`](https://gateway.pinata.cloud/ipfs/QmWLdyFMUugMtKZs1xeJCSUKerWd9M627gxjAtp6TLrAgP) to view the site. If we change `string_2` to `IPFS` then the CID of the website changes to `Qme1A6ofTweQ1JSfLLdkoehHhpbAAk4Z2hWjyNC7YJF9m5`. Now users can go to [`example.com/Qme1A6ofTweQ1JSfLLdkoehHhpbAAk4Z2hWjyNC7YJF9m5`](https://gateway.pinata.cloud/ipfs/Qme1A6ofTweQ1JSfLLdkoehHhpbAAk4Z2hWjyNC7YJF9m5).
+
+Having a user visit the site using the CID is cumbersome since the CID will change every time a variable is updated. So instead, we can use a _pointer_ that maintains the CID of the page with the latest update. This way, users can go to `example.com`, and always be directed to the latest content. This pointer is _mutable_; it can be updated to reflect the changes downstream.
```

Comment:
This is a great example. What you are describing here is in fact [IPNS](https://docs.ipfs.io/concepts/ipns). I think it's worth mentioning it as it can very easily be related to the immutability concept discussed here, adding more value to it.

This example is especially valuable because the usual selling point of IPNS is just that it saves the user from the cumbersome task of dealing with CIDs. But it's not just that CIDs are ugly and hard to remember, more importantly, they change with the content (because they _are_ the content) whereas the URL (the _Pointer_ in our examples) stays the same (as it is able, unlike CIDs, to mutate its own content: the CID/web page being updated).